translator/ml_translatorLib.sml

```diff
@@ -2713,6 +2713,35 @@ fun dest_word_shift tm =
   if wordsSyntax.is_word_ror tm then Eval_word_ror else
     failwith("not a word shift")
 
+(* CakeML signature generation and manipulation *)
+val generate_sigs = ref false;
+
+fun sig_of_mlname name = definition (ml_progLib.pick_name name ^ "_sig") |> concl |> rhs;
+
+fun module_signatures names = listSyntax.mk_list(map sig_of_mlname names, ``:spec``);
```

Comment:
Can you replace ``:spec`` with something that does not call the parser? Maybe that type is already in `astSyntax` or something?

Reply:
sure thing! I'll get onto that tomorrow :)